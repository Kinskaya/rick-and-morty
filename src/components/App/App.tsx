import React from 'react';
import { Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';

import HomePage from '../../pages/Home/HomePage';
import AboutPage from '../../pages/About/AboutPage';
import NotFoundPage from '../../pages/NotFound/NotFoundPage';
import OrderForm, { TOrderForm } from '../../pages/OrderForm/OrderForm';
import Layout from '../Layout/Layout';
import { ICard } from '../Card/Card';
import CardPage from '../../pages/CardPage/CardPage';
import { store } from '../../redux/store';

export interface IState {
  info: ICard[];
  searchValue: string | undefined;
  loading: boolean;
  error: string;
  modal: boolean;
  item: TItem;
  selectValue: string;
  orders: TOrderForm[];
  order: TOrderForm;
  currentPage: number;
  cardsPerPage: number;
  totalCards: number;
  totalPages: number;
}

export type TItem = {
  id: number;
  name: string;
  gender: string;
  image: string;
  status: string;
  species: string;
};

export interface IAction {
  type: string;
  payload?: unknown;
}

const App = () => {
  return (
    <>
      <Provider store={store}>
        <Routes>
          <Route path="/" element={<Layout />}>
            <Route index element={<HomePage />} />
            <Route path="about" element={<AboutPage />} />
            <Route path="order" element={<OrderForm />} />
            <Route path=":id" element={<CardPage />} />
            <Route path="*" element={<NotFoundPage />} />
          </Route>
        </Routes>
      </Provider>
    </>
  );
};

export default App;
