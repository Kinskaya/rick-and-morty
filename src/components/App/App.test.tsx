import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import App from '../App/App';

describe('App component', () => {
  it('render correctly', () => {
    const { asFragment } = render(<App />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
  it('render home link', () => {
    const { getByText } = render(<App />, { wrapper: BrowserRouter });
    const linkElement = getByText(/home/i);
    expect(linkElement).toBeInTheDocument();
  });
});
