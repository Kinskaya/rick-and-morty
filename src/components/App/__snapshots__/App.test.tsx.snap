// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render correctly 1`] = `
<DocumentFragment>
  <header>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Home
    </a>
    <a
      class=""
      href="/about"
    >
      About Us
    </a>
    <a
      class=""
      href="/order"
    >
      Order
    </a>
  </header>
  <main
    class="container"
  >
    <div
      class="wrapper"
    >
      <h1>
        Rick and Morty
      </h1>
      <form>
        <i
          class="material-icons"
        >
          search
        </i>
        <input
          class="search-input"
          placeholder="Search..."
          type="text"
          value=""
        />
      </form>
      <div
        class="card-container"
      />
      <div
        class="modal"
      >
        <div
          class="modal-content"
        >
          <div
            class="card"
          >
            <img
              alt="item-image"
              class="card-image"
              src=""
            />
            <p
              class="title"
            >
              Name: 
              <span />
            </p>
            <p
              class="title"
            >
              Gender: 
              <span />
            </p>
            <p
              class="title"
            >
              Status: 
              <span />
            </p>
            <p
              class="title"
            >
              Species: 
              <span />
            </p>
          </div>
          <button
            class="modal-btn"
          >
            x
          </button>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
